source/client-side-encryption/client-side-encryption.md

```diff
@@ -0,0 +1,2475 @@
+   accessKeyId: string;
+   secretAccessKey: string;
+   sessionToken?: string; // Required for temporary AWS credentials.
+};
+
+type AzureKMSOptions = AzureKMSCredentials | AzureAccessToken;
+
+interface AzureKMSCredentials {
+   tenantId: string;
+   clientId: string;
+   clientSecret: string;
+   identityPlatformEndpoint?: string; // Defaults to login.microsoftonline.com
+};
+
+interface AzureAccessToken {
+   accessToken: string;
+};
+
+type GCPKMSOptions = GCPKMSCredentials | GCPKMSAccessToken
+
+interface GCPKMSCredentials {
+   email: string;
+   privateKey: byte[] | string; // May be passed as a base64 encoded string.
+   endpoint?: string; // Defaults to oauth2.googleapis.com
+};
+
+interface GCPKMSAccessToken {
+   accessToken: string;
+}
+
+interface LocalKMSOptions {
+   key: byte[96] | string; // The master key used to encrypt/decrypt data keys. May be passed as a base64 encoded string.
+};
+
+interface KMIPKMSOptions {
+   endpoint: string;
+};
+```
+
+The following shows an example object of `KMSProviders`:
+
+```yaml
+{
+   # Pass credentials for AWS:
+   "aws": { "accessKeyId": "foo", "secretAccessKey": "bar" },
+   # Use an empty document to enable "Automatic Credentials" for Azure:
+   "azure": {},
+   # Pass an access token for GCP:
+   "gcp": { "accessToken": "foo" },
+   # Pass a 96 byte base64 encoded string for the local KMS provider.
+   "local": { "key": "Mng0NCt4ZHVUYUJCa1kxNkVyNUR1QURhZ2h2UzR2d2RrZzh0cFBwM3R6NmdWMDFBMUN3YkQ5aXRRMkhGRGdQV09wOGVNYUMxT2k3NjZKelhaQmRCZGJkTXVyZG9uSjFk" }
+   # Pass the endpoint for KMIP:
+   "kmip": { "endpoint": "localhost:5698" }
+   # Pass credentials for a different AWS account by appending a name.
+   # Note: credentials with a name do not support "Automatic Credentials".
+   "aws:name2": { "accessKeyId": "foo2", "secretAccessKey": "bar2" }
+}
+```
+
+##### Automatic Credentials
+
+Certain values of [KMSProviders](#kmsproviders) indicate a request by the user that the associated KMS providers should
+be populated lazily on-demand. The driver MUST be able to populate the respective options object on-demand
+if-and-only-if such respective credentials are needed. The request for KMS credentials will be indicated by
+[libmongocrypt](#libmongocrypt) only once they are needed.
+
+When such a state is detected, [libmongocrypt](#libmongocrypt) will call back to the driver by entering the
+`MONGOCRYPT_CTX_NEED_KMS_CREDENTIALS` state, upon which the driver should fill in the KMS options automatically.
+
+Automatic credentials are only supported for the KMS provider types `aws`, `gcp`, and `azure`. KMS providers containing
+a name (e.g. `aws:myname`) do not support automatic credentials. Attempting to configure a KMS provider with a name for
+automatic credentials results in a runtime error from [libmongocrypt](#libmongocrypt). See
+[Why do on-demand KMS credentials not support named KMS providers?](#why-do-on-demand-kms-credentials-not-support-named-kms-providers)
+
+> \[!NOTE\]
```

Comment:
The escaping here prevents it from rendering as a note tooltip.

Reply:
I filed https://github.com/KyleKing/mdformat-admon/issues/19

Reply:
Read that thread. Is there no way to remove this escaping since it would break the CI checks?

I thought you addressed this in previous RST-to-MD PRs.

Reply:
The auto-formatter is changing it to the current form.